### DISCUSSION TITLE
Parallelising task with GNU parallel

### DISCUSSION THREAD

**Discussion by leana8959** [lagmoellertim/unsilence#153] [Q&A]
Hello,

I'm trying to batch convert the files of a course.\
I tried to use this command `parallel 'unsilence -y {} unsilenced_"sourceDir"/{/}' ::: "$sourceDir"/*.mp4` to unsilence all the files in the `$sourceDir` directory.

I noticed that this command would fail after the first task, outputting "FileNotFoundError: [Errno 2] No such file or directory: '/root/.tmp/a5385abe-c716-4b0e-a7e6-cc7f0e5af6f9/concat_list.txt". To try to remedy this, I used the `--delay 2` in hope to avoid starting all tasks at the same time, so that each task can have their own `.tmp` directory. This is proven to not work, since I still get the same error message, and they don't share the same path. This means that for some reason unsilence can't find its `.tmp` even though the `.tmp` path doesn't coincide with other parallelized tasks.

Does anyone know how to solve this?

Thank you, I appreciate greatly this project :)

**Comment by bojidar-bg** [comment]
Probably caused by this line here: https://github.com/lagmoellertim/unsilence/blob/f1133363f29c0ef36a2241339eef61a666951919/unsilence/Unsilence.py#L114

If you pass different values for the `temp_dir` parameter of the `Unsilence` class it should in theory work. (:
